### PR TITLE
Fix for issue #493

### PR DESCRIPTION
**Pull request by mbudiu-vmw**

### CODE REVIEW

**Comment by mbudiu-vmw** [comment]
unfortunately it is not easy to create a negative test for this using our default testing framework, since this test succeeds with p4test, and fails with p4c-bm2-ss, and our scripts always run both compilers.

File: backends/bmv2/midend.cpp [hunk 1]
```diff
@@ -121,8 +121,11 @@ MidEnd::MidEnd(CompilerOptions& options) {
                 // nothing further to do
                 return nullptr;
             // We save the names of some control blocks for special processing later
-            if (main->getConstructorParameters()->size() != 6)
-                ::error("%1%: Expected 6 arguments for main package", main);
+            if (main->getConstructorParameters()->size() != 6) {
+                ::error("%1%: Expected 6 arguments for main package; are you using %2%?",
+                        main, P4V1::V1Model::instance.file.toString());
+                return nullptr;
```
Comment (ChrisDodd) [inline backends/bmv2/midend.cpp]: Isn't this caught by typechecking?   The constructor call to main must match the declaration in v1model.p4...

**Comment by mbudiu-vmw** [comment]
This fails when v1model was not used.
We have no easy way to check which model was used, this is an approximation of a check.

**Comment by mbudiu-vmw** [comment]
Travis has picked-up a new version of BMv2 which is very strict, checking for issue #486, which won't be fixed very soon.

**Pull request merged**